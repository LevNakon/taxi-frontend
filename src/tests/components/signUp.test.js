import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import SignUp from '../../components/signUp';

describe('SignUp --- test render correctly component', () => {
    const initialState = {
        signUpState: {
            message: '',
            success: true,
            userId: 1
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<SignUp store={store} />);
    });

    it('render correctly component SignUp', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('success')).toEqual(initialState.signUpState.success);
        expect(wrapper.dive().prop('message')).toEqual(initialState.signUpState.message);
        expect(wrapper.dive().prop('userId')).toEqual(initialState.signUpState.userId);
     });
});