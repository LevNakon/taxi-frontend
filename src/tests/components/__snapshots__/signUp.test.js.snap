// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp --- test render correctly component render correctly component SignUp 1`] = `
<ContextProvider
  value={null}
>
  <SignUp
    message=""
    signUpNull={[Function]}
    signUpWatcher={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    success={true}
    userId={1}
  />
</ContextProvider>
`;
